option/option.go

```diff
@@ -0,0 +1,250 @@
+	openapiParam := &openapi3.Parameter{
+		Name:        name,
+		In:          string(param.Type),
+		Description: param.Description,
+		Schema:      openapi3.NewStringSchema().NewRef(),
+	}
+	if param.GoType != "" {
+		openapiParam.Schema.Value.Type = &openapi3.Types{param.GoType}
+	}
+	openapiParam.Schema.Value.Nullable = param.Nullable
+	openapiParam.Schema.Value.Default = panicsIfNotCorrectType(openapiParam, param.Default)
+
+	if param.Required {
+		openapiParam.Required = param.Required
+	}
+	for name, exampleValue := range param.Examples {
+		if openapiParam.Examples == nil {
+			openapiParam.Examples = make(openapi3.Examples)
+		}
+		exampleOpenAPI := openapi3.NewExample(name)
+		exampleOpenAPI.Value = panicsIfNotCorrectType(openapiParam, exampleValue)
+		openapiParam.Examples[name] = &openapi3.ExampleRef{Value: exampleOpenAPI}
+	}
+
+	return func(r *fuego.BaseRoute) {
+		r.Operation.AddParameter(openapiParam)
+		if r.Params == nil {
+			r.Params = make(map[string]fuego.OpenAPIParam)
+		}
+		r.Params[name] = param
+	}
+}
+
+// Tags adds one or more tags to the route.
+func Tags(tags ...string) func(*fuego.BaseRoute) {
+	return func(r *fuego.BaseRoute) {
+		r.Operation.Tags = append(r.Operation.Tags, tags...)
+	}
+}
+
+// Summary adds a summary to the route.
+func Summary(summary string) func(*fuego.BaseRoute) {
+	return func(r *fuego.BaseRoute) {
+		r.Operation.Summary = summary
+	}
+}
+
+// Description adds a description to the route.
+func Description(description string) func(*fuego.BaseRoute) {
+	return func(r *fuego.BaseRoute) {
+		r.Operation.Description = description
+	}
+}
+
+// OperationID adds an operation ID to the route.
+func OperationID(operationID string) func(*fuego.BaseRoute) {
+	return func(r *fuego.BaseRoute) {
+		r.Operation.OperationID = operationID
+	}
+}
+
+// Deprecated marks the route as deprecated.
+func Deprecated() func(*fuego.BaseRoute) {
+	return func(r *fuego.BaseRoute) {
+		r.Operation.Deprecated = true
+	}
+}
+
+// AddError adds an error to the route.
+func AddError(code int, description string, errorType ...any) func(*fuego.BaseRoute) {
+	return func(r *fuego.BaseRoute) {
+		addResponse(r.MainRouter, r.Operation, code, description, errorType...)
+	}
+}
+
+func addResponse(s *fuego.Server, operation *openapi3.Operation, code int, description string, errorType ...any) {
+	var responseSchema fuego.SchemaTag
+
+	if len(errorType) > 0 {
+		responseSchema = fuego.SchemaTagFromType(s, errorType[0])
+	} else {
+		responseSchema = fuego.SchemaTagFromType(s, fuego.HTTPError{})
+	}
+	content := openapi3.NewContentWithSchemaRef(&responseSchema.SchemaRef, []string{"application/json"})
+
+	response := openapi3.NewResponse().
+		WithDescription(description).
+		WithContent(content)
+
+	operation.AddResponse(code, response)
+}
+
+// RequestContentType sets the accepted content types for the route.
+// By default, the accepted content types is */*.
+func RequestContentType(consumes ...string) func(*fuego.BaseRoute) {
```

Comment:
I don't believe this will produce a proper result. I remember having to do this in specific way.  

I'll branch off and show an example.

Reply:
Take a look at https://github.com/go-fuego/fuego/pull/193/files

You will see that I'm declaring two different request content types. For each route. But in this case both routes end up referencing the same requestBody. The route will have to only reference the schema in this case. Hence why the logic for generating this bit of the route occurred here: https://github.com/go-fuego/fuego/pull/177/files#diff-127c9bd6b06d3b470c2b0f10ec1c538b7798d9fca66d4812d554a0f3a15fbf92R97

Reply:
As discussed earlier not having access to the `RequestBody` is very unfortunate. But I guess now. The best way to do this will if to check if `if r.AcceptedContentTypes !=nil` and branch into a new code path. Could probably just reuse the code that is already in the old `RequestContentType`.

FYI I do plan on adding the server wide default option once this is done. Hence why I'm so fixated on this 😄

Reply:
Mmh I see

Reply:
> FYI I do plan on adding the server wide default option once this is done. Hence why I'm so fixated on this 😄

Veeery good idea. For the moment, we'll ship the variadic options without this feature because I want to move forward, but I'll raise an issue to track this